zio-http/src/test/scala/zio/http/api/ServerClientIntegrationSpec.scala

```diff
@@ -0,0 +1,57 @@
+package zio.http.api
+
+import zio.http.{Client, ClientConfig, Server, ServerConfig, URL}
+import zio.schema.{DeriveSchema, Schema}
+import zio.test.{ZIOSpecDefault, assertTrue}
+import zio.{Scope, ZIO, ZLayer}
+
+object ServerClientIntegrationSpec extends ZIOSpecDefault {
+
+  trait PostsService {
+    def getPost(userId: Int, postId: Int): ZIO[Any, Throwable, Post]
+  }
+
+  final case class Post(id: Int, title: String, body: String, userId: Int)
+
+  object Post {
+    implicit val schema: Schema[Post] = DeriveSchema.gen[Post]
+  }
+
+  val usersPostAPI     = API.get("users" / In.int / "posts" / In.int).out[Post]
+  val usersPostHandler = usersPostAPI.handle { case (userId, postId) =>
+    ZIO.succeed(Post(postId, "title", "body", userId))
+  }
+
+  // TODO: [Ergonomics] Need to make it easy to create an APIExecutor layer
+  def makeExecutor(client: Client) = {
+    val registry = APIRegistry.empty.registerAll(URL.fromString("http://localhost:8080").getOrElse(???)) {
+      usersPostAPI ++ usersPostAPI
+    }
+    APIExecutor(client, registry)
+  }
+
+  val executorLayer = ZLayer.fromFunction(makeExecutor _)
+
+  def spec =
+    suite("ServerClientIntegrationSpec")(
+      test("server and client integration") {
+        for {
+          _        <- Server.install(usersPostHandler.toHttpApp)
+          _        <- ZIO.debug("Installed server")
+          executor <- ZIO.service[APIExecutor[usersPostAPI.Id]]
+          // QUESTION: Do we want to encode `E` in an API?
```

Comment:
I think rather, we want to make sure the handlers can only fail with `HttpError`, and then we can wrap up any fatal error in an `InternalServerError`.

Another possibility would be to have a table in `API` that maps some `E` to `HttpError`. This would have the nice side-effect of allowing us to put error codes in docs if used properly. But probably it wouldn't be used properly since if you're failing with `HttpError` then you would just map the sum type to itself, which would not permit any kind of reasonable documentation.